### ISSUE TITLE
Weather plugin installation fails

### ISSUE THREAD

**Issue by samuel-pt** [itsabot/abot#108]
Weather plugin installation fails with below error.
I just did a fresh setup and tried to install github.com/itsabot/plugin_weather

```
>abot plugin install
Fetching 1 plugins...
# github.com/itsabot/plugin_weather
../plugin_weather/weather.go:34: undefined: dt.PluginFns
../plugin_weather/weather.go:36: too many arguments in call to plugin.New
../plugin_weather/weather.go:40: p.Vocab undefined (type *dt.Plugin has no field or method Vocab)
../plugin_weather/weather.go:40: undefined: dt.NewVocab
../plugin_weather/weather.go:41: undefined: dt.VocabHandler
../plugin_weather/weather.go:49: undefined: dt.VocabHandler
../plugin_weather/weather.go:66: p.Vocab undefined (type *dt.Plugin has no field or method Vocab)
../plugin_weather/weather.go:83: sm.SetMemory undefined (type *dt.StateMachine has no field or method SetMemory)
../plugin_weather/weather.go:115: sm.HasMemory undefined (type *dt.StateMachine has no field or method HasMemory)
../plugin_weather/weather.go:116: sm.GetMemory undefined (type *dt.StateMachine has no field or method GetMemory)
../plugin_weather/weather.go:116: too many errors
```

And below is my plugin.json content
```
{
	"Name": "Abot",
	"Description": "Digital assistant framework",
	"Version": "0.2.0-alpha",
	"ImportPath": "github.com/itsabot/abot",
	"Dependencies": {
		"github.com/itsabot/plugin_weather": "*"
	}
}

```

**Comment by egtann** [comment]
Hey @samuel-pt, ensure you have the latest version of Abot. There weather plugin is using a new API, but your Abot is looking for the old (now removed) API.

**Comment by egtann** [comment]
Should be fixed with #110. After you pull the update, you'll need to run a new migration, too. Something like this:

`psql -U postgres -d abot -a -f db/migrations/up/1463070851_drop_states_userid_not_null.sql`

Everything *should* work now, but let me know!

**Comment by samuel-pt** [comment]
Thanks a lot. 
It is working fine with the latest changes
